lib/chef/provider/package/zypper.rb

```diff
@@ -157,6 +158,14 @@ def gpg_checks
         def allow_downgrade
           "--oldpackage" if new_resource.allow_downgrade
         end
+
+        def clean_deps
+          "--clean-deps" if new_resource.action == :purge
+        end
+
+        def auto_agree_licenses
+          "--auto-agree-with-licenses" if new_resource.action == :install
+        end
```

Comment:
this is creating more complicated code by burying the conditionals down here.  i'm not sure why all of that needed to get refactored.  the `--auto-agree-with-licenses` flag should get moved back to install and the `--clean-deps` flag should get moved back to purge.

Reply:
Thank you @lamont-granquist for review. We may need to introduce other property `--global-options` for `package` resource If we don't do above changes. As `zypper` provides two options one `global-options` and other is `command-options`. Also if we introduce `--global-options` to `package` resource then this property will become redundant for other package managers like `yum` , `apt` etc. Since they provide only one option.
 
### Zypper options 
`zypper [--global-options] <command> [--command-options] [arguments]`

However I am verifying it again for `--auto-agree-with-licenses` to move it in install and `--clean-deps` to purge.

I am also looking into the ways how to make better use of introducing `--global-options` either to `package` OR to `zypper_package` resource.

Please suggest your views on this.